### PR TITLE
Moved potential side effect from assert()

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

File: lib/Remotery.c [hunk 1]
```diff
@@ -4226,10 +4226,13 @@ static Message* rmtMessageQueue_AllocMessage(rmtMessageQueue* queue, rmtU32 payl
...
 static void rmtMessageQueue_CommitMessage(Message* message, MessageID id)
 {
+    MessageID r;
     assert(message != NULL);
 
     // Setting the message ID signals to the consumer that the message is ready
-    assert(LoadAcquire((rmtU32*)&message->id) == MsgID_NotReady);
+    r = (MessageID)LoadAcquire((rmtU32*)&message->id);
+    (void)r;
+    assert(r == MsgID_NotReady);
```
Comment (JCash) [inline lib/Remotery.c:4235]: In the case someone builds with NDEBUG, removing all the asserts.

**Comment by dwilliamson** [comment]
Any updates on this?

**Comment by JCash** [comment]
Hmm, not sure what you mean?

I think it's a good rule of thumb to not include code with side effects in the assert macros, as they are removed when compiling with NDEBUG.
Although it may not be actually crucial in this case, it sets an example to others to not do it either.

**Comment by dwilliamson** [comment]
The use of `(void)r` to notify a potentially unused parameter is not compatible across compilers. That's where we can use `RMT_UNREFERENCED_PARAMETER(r)`.

**Pull request merged**